src/client/graphics/layers/BuildMenu.ts

```diff
@@ -240,12 +240,40 @@ export class BuildMenu extends LitElement implements Layer {
     const unit = this.playerActions.buildableUnits.filter(
       (u) => u.type == item.unitType,
     );
-    if (!unit) {
+    if (!unit || unit.length === 0) {
```

Comment:
```suggestion
    if (unit.length === 0) {
```
According to the interface unit/units is not nullable or optional, therefore the array will never be falsy, and the `!unit` has no use here